plugins/dicom_viewer/server/__init__.py

```diff
@@ -1,67 +1,129 @@
-        files = list(Item().childFiles(item))
-        # process files if they haven't been processed yet
-        for i, f in enumerate(files):
-            if force or 'dicom' not in f:
-                files[i] = process_file(f)
-        # filter out non-dicom files
-        files = [x for x in files if x.get('dicom')]
-        # sort files
-        files = sorted(files, key=sort_key)
-        # execute filters
-        if filters:
-            for f in files:
-                dicom = f['dicom']
-                dicom = dict((k, dicom[k]) for k in filters if k in dicom)
-                f['dicom'] = dicom
-        return files
-
-
-def sort_key(f):
-    dicom = f.get('dicom', {})
+    def makeDicomItem(self, item):
+        """Try to convert an existing item into a "DICOM item", which contains a
+        "dicomMeta" field with DICOM metadata that is common to all DICOM files.
+        """
+        metadataReference = None
+        dicomFiles = []
+
+        for file in Item().childFiles(item):
+            dicomMeta = _parseFile(file)
+            if dicomMeta is None:
+                continue
+            dicomFiles.append(_extractFileData(file, dicomMeta))
+            dicomFiles.sort(key=_getDicomFileSortKey)
+            if metadataReference is None:
+                metadataReference = dicomMeta
+            else:
+                metadataReference = _removeUniqueMetadata(metadataReference, dicomMeta)
+
+        if len(dicomFiles) is not 0:
+            # Sort the dicom files
+            dicomFiles = sorted(dicomFiles, key=_getDicomFileSortKey)
+            # Store in the item
+            item['dicom'] = {}
+            item['dicom']['meta'] = metadataReference
+            item['dicom']['files'] = dicomFiles
+            # Save the item
+            Item().save(item)
+
+
+def _extractFileData(file, dicom):
+    """Extract the usefull data to be stored in the `item['dicom']['files']`.
+    In this way it become simpler to sort them and store them.
+    """
+    return {
+        'dicom': {
+            'SeriesNumber': dicom.get('SeriesNumber'),
+            'InstanceNumber': dicom.get('InstanceNumber'),
+            'SliceLocation': dicom.get('SliceLocation')
+        },
+        'name': file['name'],
+        '_id': file['_id']
+    }
+
+
+def _getDicomFileSortKey(f):
+    """These properties are used to sort the files into the item."""
+    meta = f.get('dicom')
     return (
-        dicom.get('SeriesNumber'),
-        dicom.get('InstanceNumber'),
-        dicom.get('SliceLocation'),
-        f['name'],
+        f.get('name'),
+        meta.get('SeriesNumber'),
+        meta.get('InstanceNumber'),
+        meta.get('SliceLocation')
+    )
+
+
+def _removeUniqueMetadata(dicomMeta, additionalMeta):
+    """Return only the common data between the two inputs.
+    Only work if all the data element are hashable,
+    this means not have any dict or list as properties
+    """
+    return dict(
```

Comment:
We should eventually re-evaluate this, and see if we can do it recursively, since it only works for single-level lists here. For right now, we can leave it alone though.